### PR TITLE
WIP: guix: init at 0.15.0

### PR DESCRIPTION
**Pull request by zimbatm**
###### Motivation for this change

For fun?

guix doesn't compile yet as it's not able to find guile-git for some reason. I don't have enough guile knowledge or willpower left to debug this further.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by zimbatm** [comment]
Please close this PR in a week if nobody showed interest in this PR. 
93ff16b is the only commit that is master-worthy.

**Comment by ryantm** [comment]
@zimbatm To me, it seems quite clear that we want guix to be available; I hope you can get it to work!

**Comment by volth** [comment]
Interesting.
Sometimes their have packages newer or absent in ```nixpkgs```.

It also makes number of Guile packages big enough for a first-class support of a new language infrastructure in ```nixpkgs```

**Comment by zimbatm** [comment]
I wonder how it's all going to integrate if the package is installed by nix, I guess guix will want to also install guix with itself as well.

**Comment by domenkozar** [comment]
btw we had guix but I removed it since noone wanted to maintain it (version was very very old).

File: pkgs/development/interpreters/guile/setup-hook-2.2.sh [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
-        export GUILE_LOAD_PATH="${GUILE_LOAD_PATH}${GUILE_LOAD_PATH:+:}$1/share/guile/site"
-        export GUILE_LOAD_COMPILED_PATH="${GUILE_LOAD_COMPILED_PATH}${GUILE_LOAD_COMPILED_PATH:+:}$1/share/guile/site"
+    local ver=2.2
+    if [[ -d "$1/share/guile/site/$ver" ]]; then
+        addToSearchPath GUILE_LOAD_PATH "$1/share/guile/site/$ver"
+    elif [[ -d "$1/share/guile/site" ]]; then
+        addToSearchPath GUILE_LOAD_PATH "$1/share/guile/site"
+    fi
+
+    if [[ -d "$1/lib/guile/2.0/site-ccache" ]]; then
```
Comment (adisbladis) [inline pkgs/development/interpreters/guile/setup-hook-2.2.sh]: Typo: `2.0` -> `$ver`

File: pkgs/tools/package-management/guix/default.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    url = "https://alpha.gnu.org/gnu/guix/guix-${version}.tar.gz";
+    sha256 = "07fbjgqb5hg2b9s5z2sg1g4fqfrap5x6jxkdah0i5qbgip9p45ap";
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    pkgconfig
+  ];
+
+  buildInputs = [
```
Comment (adisbladis) [inline pkgs/tools/package-management/guix/default.nix]: `nix` is also a required buildInput

**Comment by zimbatm** [comment]
@GrahamcOfBorg build guix

File: pkgs/development/interpreters/guile/setup-hook-2.2.sh (hunk 1 above)
Comment (zimbatm) [reply pkgs/development/interpreters/guile/setup-hook-2.2.sh]: thanks, fixed

File: pkgs/tools/package-management/guix/default.nix (hunk 2 above)
Comment (zimbatm) [reply pkgs/tools/package-management/guix/default.nix]: thanks, added

**Comment by zimbatm** [comment]
> btw we had guix but I removed it since noone wanted to maintain it (version was very very old).

Yeah I don't know if this is worth it to be honest. I'll close the PR and submit another one for the guile hooks improvements.

**Comment by timokau** [comment]
This would also make it easier to tinker with and take inspiration from (read steal) guix's good usability ideas.

**Pull request closed without merging**